sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -0,0 +1,416 @@
+        public void ClientCloseError(Type clientType,
+                                     string eventHubName,
+                                     string clientId,
+                                     string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(11, clientType.Name, eventHubName ?? String.Empty, clientId ?? String.Empty, errorMessage ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that retrieval of the Event Hub properties has started.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub that properties are being retrieved for.</param>
+        ///
+        [Event(12, Level = EventLevel.Informational, Message = "Retrieving properties for Event Hub: {0} (Partition Id: '{1}').")]
+        public void GetPropertiesStart(string eventHubName)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(12, eventHubName ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that retrieval of the Event Hub properties has completed.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub that properties are being retrieved for.</param>
+        ///
+        [Event(13, Level = EventLevel.Informational, Message = "Completed retrieving properties for Event Hub: {0}.")]
+        public void GetPropertiesComplete(string eventHubName)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(13, eventHubName ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while retrieving Event Hub properties.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub that properties are being retrieved for.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(14, Level = EventLevel.Error, Message = "An exception occurred while retrieving properties for Event Hub: {0}. Error Message: '{2}'")]
+        public void GetPropertiesError(string eventHubName,
+                                       string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(14, eventHubName ?? String.Empty, errorMessage ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that retrieval of the Event Hub partition properties has started.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub that properties are being retrieved for.</param>
+        /// <param name="partitionId">The identifier of the partition hat properties are being retrieved for.</param>
```

Comment:
Typo: hat (there are other occurrences in the same file).

Reply:
Got 'em.   Yay, copy/paste!